### PR TITLE
feat(metrics): Add percentiles setting

### LINKED ISSUE #71515: Ability to enable percentiles
turning on percentiles are going to slow down your queries, and will eventually result in higher bill and it's an org thing and distributions only

### PR DESCRIPTION
**Pull request by ArthurKnaus**
Add setting for enabling custom metric percentiles.
Add disabled percentile entries to op select.

https://github.com/getsentry/sentry/assets/7033940/85d7a8bf-1c35-4ac2-a616-0a00169edc99

https://github.com/getsentry/sentry/assets/7033940/870910df-2f55-49c5-8266-a376eac435f8

Requires https://github.com/getsentry/sentry/pull/71554
Closes https://github.com/getsentry/sentry/issues/71515

### CODE REVIEW

File: static/app/utils/useOverlay.tsx [hunk 1]
```diff
@@ -263,6 +263,9 @@ function useOverlay({
       shouldCloseOnBlur,
       isKeyboardDismissDisabled,
       shouldCloseOnInteractOutside: target => {
+        if (target && target.getAttribute('data-allow-overlay-interaction') === 'true') {
```
Comment (anonrig) [inline static/app/utils/useOverlay.tsx:266]:
```suggestion
        if (target?.getAttribute('data-allow-overlay-interaction') === 'true') {
```

File: static/app/views/metrics/queryBuilder.tsx [hunk 2]
```diff
@@ -268,6 +295,31 @@ export const QueryBuilder = memo(function QueryBuilder({
...
   );
 
+  const opOptions = useMemo(() => {
+    const options: SelectOption<MetricsAggregate>[] =
+      selectedMeta?.operations.filter(isAllowedOp).map(op => ({
+        label: op,
+        value: op,
+      })) ?? [];
+
+    const firstProject =
```
Comment (anonrig) [inline static/app/views/metrics/queryBuilder.tsx:305]: firstProjst can be undefined since `selectedProjects[0]` can return undefined

**Review by priscilawebdev — APPROVED** [review]
You might want to wait for a second approval from someone more familiar with the current status of metrics, but the frontend code looks good to me

**Comment by ArthurKnaus** [comment]
Not planned

**Pull request closed without merging**